tests/remotes/base.py

```diff
@@ -1,7 +1,61 @@
+import pathlib
+
 from funcy import cached_property
 
+from dvc.path_info import URLInfo
+
+
+class Base(URLInfo):
+    def is_file(self):
+        raise NotImplementedError
+
+    def is_dir(self):
+        raise NotImplementedError
+
+    def exists(self):
+        raise NotImplementedError
+
+    def mkdir(self, mode=0o777, parents=False, exist_ok=False):
+        raise NotImplementedError
+
+    def write_text(self, contents, encoding=None, errors=None):
+        raise NotImplementedError
+
+    def write_bytes(self, contents):
+        raise NotImplementedError
+
+    def read_text(self, encoding=None, errors=None):
+        raise NotImplementedError
+
+    def read_bytes(self):
+        raise NotImplementedError
+
```

Comment:
All of these are your regular pathlib.Path's methods.

Reply:
For now we have a bare minimum, but it is easy to extend with what we need in the future.